src/utils/content/index.ts

```diff
@@ -131,13 +133,17 @@ export async function createParser(collection: ResolvedCollection, nuxt?: Nuxt)
     if (file.path && !file.dirname) {
       file.dirname = dirname(file.path)
     }
+    const beforeParseCtx: FileBeforeParseHook = { file, collection, parserOptions }
+    // @ts-expect-error runtime type
+    await nuxt?.callHook?.('content:file:beforeParse', beforeParseCtx)
+    const { file: hookedFile, collection: hookedCollection } = beforeParseCtx
 
-    const parsedContent = await transformContent(file, parserOptions)
+    const parsedContent = await transformContent(hookedFile, beforeParseCtx.parserOptions)
     const { id: id, ...parsedContentFields } = parsedContent
     const result = { id } as typeof collection.extendedSchema._type
     const meta = {} as Record<string, unknown>
 
-    const collectionKeys = Object.keys(collection.extendedSchema.shape)
+    const collectionKeys = Object.keys(hookedCollection.extendedSchema.shape)
```

Comment:
```suggestion
    const collectionKeys = Object.keys(collection.extendedSchema.shape)
```

it's more secure to use `collection` rather than `hookedCollection`. Collections in these hooks should be readonly

Reply:
I wasn't exactly sure how all of my modules integrating with Nuxt Content v3 would work so I wanted to be able to modify a collection schema using a hook if needed.

Thinking on your comment, I now believe this would be better done via a dedicated hook if it is needed. I can send a PR if / when that blocks me.